docs/eeeoh.md

```diff
@@ -0,0 +1,460 @@
+We may investigate codemods for existing code so we can remove this compatibility layer in a future major version.
+
+### `x-request-id`
+
+See [RFC002] for more information.
+
+## Datadog log tiers
+
+[Datadog tiering] allows applications to choose the appropriate feature set and retention period for their logs.
+
+`@seek/logger` supports multiple approaches for configuring Datadog log tiers.
+
+### Tier default
+
+The simplest approach is to specify a static default tier:
+
+```typescript
+import { createLogger } from '@seek/logger';
+
+const logger = createLogger({
+  base,
+  eeeoh: {
+    datadog: 'tin',
+  },
+});
+
+logger.info('A tin message');
+logger.warn('A tin message');
+logger.error('A tin message');
+```
+
+### Tier by level
+
+To retain warnings for longer than informational logs,
+you may specify level-based tiering:
+
+```typescript
+import { createLogger } from '@seek/logger';
+
+const logger = createLogger({
+  base,
+  eeeoh: {
+    datadog: [
+      // Default
+      'tin',
+      {
+        // All levels below warn will inherit `tin`
+        warn: 'silver',
+        // All levels above warn will inherit `silver`
+      },
+    ],
+  },
+});
+
+logger.info('A tin message');
+logger.warn('A silver message');
+logger.error('A silver message');
+```
+
+### Tier override
+
+Consider an application with logs where a `tin` default is generally fine.
+However, it has a particular audit log that requires longer retention.
+
+First, define the default:
+
+```typescript
+const logger = createLogger({
+  base,
+  eeeoh: { datadog: 'tin' },
+});
+```
+
+Then, override the tier on a particular audit log:
+
+```typescript
+logger.info({ eeeoh: { datadog: 'tin-plus' } }, 'A silver message');
+```
+
+You may also selectively opt out of routing a particular log:
+
+```typescript
+logger.info({ eeeoh: { datadog: false } }, 'A message sent into the void');
+```
+
+## Multiple loggers
+
+### By service
+
+Some codebases may share logger code across multiple components.
+However, each component requires a distinct `service` value.
+
+We recommend varying `service` via environment variable:
+
+```yaml
+functions:
+  FunctionA:
+    environment:
+      SERVICE: component-a
+
+  FunctionB:
+    environment:
+      SERVICE: component-b
+```
+
+```typescript
+import { Eeeoh } from '@seek/logger';
+import { Env } from 'skuba-dive';
+
+const base = {
+  env: Env.oneOf(Eeeoh.envs)('DD_ENV'),
+  service: Env.string('SERVICE'), // 'component-a' | 'component-b'
+  version: Env.string('DD_VERSION'),
+} as const;
+```
+
+However,
+it's also possible to specify [child loggers] that derive from a common root logger:
+
+```typescript
+// Do not export so it is not used directly
+const _logger = createLogger({
+  base,
+  eeeoh: { datadog: 'tin' },
+});
+
+export const aLogger = _logger.child({ service: 'component-a' });
+
+export const bLogger = _logger.child({ service: 'component-b' });
+```
```

Comment:
Considering removing this documented alternative (and possibly banning `service` on child loggers) if we don't see a good use case for it

Reply:
Surely if it's a new service it's not that much more template to just define a new logger per service?
And if people really want to reduce config they can do something like: 
```typescript
const loggerConfig = {
  base,
  eeeoh: { datadog: 'tin' },
  etc...
};

const aLogger = createLogger({
  ...loggerConfig,
  service: 'component-a'
});

const bLogger = createLogger({
  ...loggerConfig,
  service: 'component-b'
});